File: C#/TemplateMethod/PdfCVReportGeneration.cs
using dotnet.TemplateMethod.Models;

namespace dotnet.TemplateMethod;

public class PdfCVReportGeneration : CvReportGeneration
{
    protected override Data ExtractData(Uri cvPath)
    {
        Console.WriteLine("Extracting data from pdf...");
        
        return new Data();
    }
}

File: typescript/patterns/template/pdf-cv-report-generation.ts
import { CVReportGeneration } from "./cv-report-generation";
import { Data } from "./data";
import { FileInfo } from "./file-info";

export class PdfCVReportGeneration extends CVReportGeneration {
  override extractData(file: FileInfo): Data {
    console.log("extracting data from PDF");
    return new Data();
  }
}
